runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3136,45 +3132,69 @@ remoteCompile(
       serializeClass = compInfo->getclassesCachedAtServer().insert(clazz).second;
       }
 
-   auto classInfoTuple = JITServerHelpers::packRemoteROMClassInfo(clazz, compiler->fej9vm()->vmThread(), compiler->trMemory(), serializeClass);
+   auto classInfoTuple = JITServerHelpers::packRemoteROMClassInfo(clazz, compiler->fej9vm()->vmThread(),
+                                                                  compiler->trMemory(), serializeClass);
    std::string optionsStr = TR::Options::packOptions(compiler->getOptions());
-   std::string recompMethodInfoStr = compiler->isRecompilationEnabled() ? std::string((char *) compiler->getRecompilationInfo()->getMethodInfo(), sizeof(TR_PersistentMethodInfo)) : std::string();
+   std::string recompMethodInfoStr = compiler->isRecompilationEnabled()
+      ? std::string((const char *)compiler->getRecompilationInfo()->getMethodInfo(), sizeof(TR_PersistentMethodInfo))
+      : std::string();
+
+   uintptr_t *classChain = NULL;
+   std::vector<J9Class *> ramClassChain;
+   std::vector<J9Class *> uncachedRAMClasses;
+   std::vector<JITServerHelpers::ClassInfoTuple> uncachedClassInfos;
+   if (aotCacheStore)
+      {
+      classChain = compiler->fej9vm()->sharedCache()->rememberClass(clazz);
+      if (classChain)
+         {
+         uintptr_t len = classChain[0] / sizeof(classChain[0]) - 1;
```

Comment:
Why is the `-1` needed?

Reply:
`classChain[0]` stores the size of the whole class chain record in bytes, including the 1st word itself. `len` is the number of classes in the chain.